app/controllers/hiring_staff/vacancies/documents_controller.rb

```diff
@@ -52,11 +52,16 @@ def documents_form_params
 
   def redirect_if_no_supporting_documents
     supporting_documents = session[:vacancy_attributes]['supporting_documents']
-    redirect_to supporting_documents_school_job_path unless supporting_documents == 'yes'
+    redirect_to supporting_documents_school_job_path unless supporting_documents
+  end
+
+  def next_step
+    application_details_school_job_path
   end
 
   def redirect_to_next_step_if_save_and_continue
-    redirect_to application_details_school_job_path if params[:commit] == 'Save and continue'
+    vacancy = Vacancy.find(session[:vacancy_attributes]['id'])
```

Comment:
This line is redundant as we have a method called `vacancy` that does that

Reply:
Got rid of the line